fastapi/models/fastapi_endpoint.py

```diff
@@ -85,6 +92,11 @@ def _compute_urls(self):
             rec.redoc_url = f"{rec.root_path}/redoc"
             rec.openapi_url = f"{rec.root_path}/openapi.json"
 
+    @api.depends("user_id")
+    def _compute_company_id(self):
+        for endpoint in self:
+            endpoint.company_id = endpoint.user_id.company_id
```

Comment:
A little style remark: I think `for rec in self` or `for record in self`, when done consistently is easier to read than `for endpoint in self`. Just like self is... self, `rec` or `record`  for the current record in the self iterator.